File: modal-rb/examples/sandbox.rb
require 'modal'

def main
  app = Modal::App.lookup("my-modal-app")
  puts "Found app: #{app.app_id}"
  image = app.image_from_registry("alpine:latest")
  puts "Using image: #{image.image_id}"
  sandbox = app.create_sandbox(image, timeout: 30000, cpu: 0.5, memory: 256, command: ["cat"])
  puts "Created sandbox with ID: #{sandbox.sandbox_id}"
  sandbox.stdin.write("Hello from Modal!")
  sandbox.stdin.close
  output = sandbox.stdout.read
  puts "Sandbox output: #{output}"
end

main

File: modal-js/examples/sandbox.ts
import { ModalClient } from "modal";

const mc = new ModalClient();

const app = await mc.apps.lookup("libmodal-example", {
  createIfMissing: true,
});
const image = mc.images.fromRegistry("alpine:3.21");

const sb = await mc.sandboxes.create(app, image, { command: ["cat"] });
console.log("Sandbox:", sb.sandboxId);

const sbFromId = await mc.sandboxes.fromId(sb.sandboxId);
console.log("Queried Sandbox from ID:", sbFromId.sandboxId);

await sb.stdin.writeText("this is input that should be mirrored by cat");
await sb.stdin.close();
console.log("output:", await sb.stdout.readText());

await sb.terminate();
